gui/src/main.rs

```diff
@@ -39,37 +39,15 @@ pub enum RequestWindowMode {
     ENVIRONMENT,
 }
 
-#[derive(Serialize, Deserialize)]
-pub struct GuiConfig {
-    pub active_window: ActiveWindow,
-    pub request_window_mode: RequestWindowMode,
-    pub selected_http_method: HttpMethod,
-    pub url: String,
-    pub body_str: String,
-    pub import_window_open: bool,
-    pub import_mode: ImportMode,
-    pub import_file_path: String,
-}
-impl Default for GuiConfig {
-    fn default() -> Self {
-        Self {
-            active_window: ActiveWindow::REQUEST,
-            request_window_mode: RequestWindowMode::BODY,
-            selected_http_method: HttpMethod::GET,
-            url: String::from("{{HOST_URL}}/json"),
-            body_str: String::from("{ \"foo\": \"bar\" }"),
-            import_window_open: false,
-            import_file_path: String::from(""),
-            import_mode: ImportMode::COLLECTION,
-        }
-    }
-}
-
 pub struct Gui {
     pub response: Arc<RwLock<Option<Value>>>,
     pub headers: Rc<RefCell<Vec<(bool, String, String)>>>,
     pub selected_environment: Rc<RefCell<Option<api::domain::environment::EnvironmentFile>>>,
     pub environments: Rc<RefCell<Option<Vec<api::domain::environment::EnvironmentFile>>>>,
+    pub request_history_items: Rc<RefCell<Option<Vec<api::domain::request_item::RequestHistoryItem>>>>,
+    pub selected_history_item: Rc<RefCell<Option<api::domain::request_item::RequestHistoryItem>>>,
+    pub saved_requests: Rc<RefCell<Option<HashMap<String, api::domain::request::DBRequest>>>>,
```

Comment:
There may be a more efficient way but I figured the easiest way would be to load all saved requests and responses from the db, and then put them in a HashMap of their id to the contents. That way, when the RequestHistoryItems are retrieved, the id's can be swapped for full info to update the UI based on the two ids.